src/server/location.rs

```diff
@@ -31,10 +31,21 @@ impl From<&RequestHead> for LocationResult {
         let headers = head.headers();
         if let Some(loc_string) = headers.get(GOOG_LOC_HEADER) {
             dbg!("Found google header", loc_string);
-            let mut parts: Vec<&str> = loc_string.to_str().unwrap_or("").split(',').collect();
+            let mut parts = loc_string.to_str().unwrap_or("").split(',');
+            let country = parts.next().map(ToOwned::to_owned);
+            let subdivision = parts.next().map(|subdivision| {
+                // client_region_subdivision: a "Unicode CLDR subdivision ID,
+                // such as USCA or CAON"
+                if subdivision.len() < 3 {
+                    subdivision
```

Comment:
I wonder if we'd ever hit this branch as according to the doc the first two letters should always be the ISO-3166-2 country code. Just be safe ;)

Reply:
*mumble*mumble* never trust outside data to be consistent *mumble*mumble*